apps/website/src/components/forms/EntryForm.tsx

```diff
@@ -55,92 +58,182 @@ export const EntryForm: React.FC<{
+              <tbody>
+                <tr>
+                  <th className="text-left">Username</th>
+                  <td>{session.user.name}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">Email</th>
+                  <td>{entry.email}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">Name</th>
+                  <td>
+                    {entry.givenName} {entry.familyName}
+                  </td>
+                </tr>
+
+                <tr>
+                  <th colSpan={2} className="pt-2 text-left text-xl">
+                    Shipping address
+                  </th>
+                </tr>
+                <tr>
+                  <th className="text-left">Street address</th>
+                  <td>{entry.addressLine1}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">Second address line</th>
+                  <td>{entry.addressLine2}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">City</th>
+                  <td>{entry.city}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">State / Province / Region</th>
+                  <td>{entry.state}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">ZIP / Postal Code</th>
+                  <td>{entry.postalCode}</td>
+                </tr>
+                <tr>
+                  <th className="text-left">Country</th>
+                  <td>{entry.country && getCountryName(entry.country)}</td>
+                </tr>
+              </tbody>
+            </table>
+          </Section>
         </>
       )}
 
-      <Headline>Enter your details</Headline>
-
-      <div className="flex flex-col gap-4">
-        <NameFieldset />
-        <ContactFieldset
-          defaultEmailAddress={session.user.email || undefined}
-        />
-        <ShippingAddressFieldset />
-        {config.hasRules && <EntryRulesFieldset form={form} />}
-
-        <ConsentFieldset />
-      </div>
-
-      <div className="mt-7">
-        <Button type="submit" disabled={enterForm.isLoading}>
-          {config.submitButtonText || "Enter to Win"}
-        </Button>
-      </div>
-    </form>
+      {/* Handle users that are logged in, but haven't submitted yet */}
+      {session?.user?.id && !entry && (
+        <form onSubmit={handleSubmit}>
+          {config.checks && (
+            <>
+              <Section containerClassName="max-w-lg">
+                {/* Surface errors at the start */}
+                {enterForm.error && (
+                  <MessageBox variant="failure">
+                    Error: {enterForm.error.message}
+                  </MessageBox>
+                )}
+
+                <Heading level={2} className="mb-6 text-2xl">
+                  Steps to Enter
```

Comment:
To make the split form easier to understand, can add numbers to the parts? I'm using actions instead of steps so its less confusing.

```suggestion
                  1) Actions to Enter
```